internal/stanza/converter.go

```diff
@@ -15,49 +15,276 @@
 package stanza
 
 import (
+	"context"
+	"encoding/json"
 	"fmt"
+	"sync"
+	"time"
 
 	"github.com/open-telemetry/opentelemetry-log-collection/entry"
 	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
 )
 
-// Convert a stanza-style entry to a pdata.Logs
-func Convert(obsLog *entry.Entry) pdata.Logs {
+// Converter converts entry.Entry into pdata.Logs aggregating translated
+// entries into logs coming from the same Resource.
+// Logs are being sent out based on the flush interval and/or the maximum
+// batch size.
+type Converter struct {
+	// pLogsChan is a channel on which batched logs will be sent to.
+	pLogsChan chan pdata.Logs
+
+	stopOnce sync.Once
+	stopChan chan struct{}
+
+	// flushInterval defines how often we flush the aggregated log entries.
+	flushInterval time.Duration
+	// flushTriggerCount defines what's the amount of entries in the buffer that
+	// will trigger a flush of log entries.
+	flushTriggerCount uint
+	// flushChan is an internal signalling channel to trigger flushing of
+	// log entries.
+	flushChan chan struct{}
```

Comment:
To be honest I didn't think about this scenario.

A relatively easy fix for this would be to leave the separate goroutine as is but instead os signalling the flush (via `chan struct{}`) we could signal **and** send an explicit slice of logs ready for sending via `chan []pdata.Logs` (which would already be removed from the cache at this point).

Reply:
Well I can't think of a different approach to be honest.

This way you `Start()` the loop, it spins of a goroutine (as expected? `receiver` works in the same way) which will accept new entries independently of where `Batch()` is called.

How would you see running the main loop otherwise? Call to `Start()` can't be blocking, right? Or would you rather see it called as `go r.converter.Start()`?

Reply:
I think I understand now. The benefit is that the main loop is doing the minimum necessary, and so it maximally available to receive logs. This makes sense now.

Reply:
That's a nice way to sum it up 👍

Reply:
> user cannot actually place a hard limit on the max number of logs per flush, but might have to expect a few extras to join the batch before sending

I confirmed this by setting `flush_interval: 30s` in the testbed scenario. The test fails because non-uniform number of logs is consumed each batch, and the flush interval is never reached to flush the remaining logs.